src/test/java/picard/metrics/MetricBaseTest.java

```diff
@@ -0,0 +1,40 @@
+package picard.metrics;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.analysis.CollectRawWgsMetrics;
+import picard.analysis.CollectWgsMetricsWithNonZeroCoverage;
+import picard.analysis.directed.HsMetrics;
+import picard.analysis.directed.TargetMetrics;
+import picard.analysis.directed.TargetedPcrMetrics;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class MetricBaseTest {
+
+    @DataProvider(name= "testMetricClasses")
+    public Object[][] getMetricClasses(){
+        return new Object[][]{
+                {TargetedPcrMetrics.class},
+                {HsMetrics.class},
+                {TargetMetrics.class},
+                {CollectRawWgsMetrics.RawWgsMetrics.class},
+                {CollectWgsMetricsWithNonZeroCoverage.WgsMetricsWithNonZeroCoverage.class}
+            };
+        }
+
+    @Test(dataProvider = "testMetricClasses")
+    public void testUniqueFields(Class metricClass) {
+
+        Set<String> metricFields = new HashSet<>();
+
+        for (final Field f : metricClass.getFields()) {
+            Assert.assertTrue(metricFields.add(f.getName()), f.getName() + " has a naming collision");
```

Comment:
```suggestion
Assert.assertFalse(!metricFields.add(f.getName()), metricClass + " has a naming collision in field " + f.getName());
```

Reply:
In my suggestion I flipped the assertion so that the error message is clearer:
```
java.lang.AssertionError: class picard.analysis.directed.TargetedPcrMetrics has a naming collision in field GENOME_SIZE expected [false] but found [true]
```